src/app/util/attribute-storage.cpp

```diff
@@ -93,6 +93,11 @@ uint8_t singletonAttributeData[ACTUAL_SINGLETONS_SIZE];
 
 uint16_t emberEndpointCount = 0;
 
+/// Determines a incremental unique index for ember
+/// metadata that is increased whenever a structural change is made to the
+/// ember metadata (e.g. changing dynamic endpoints or enabling/disabling endpoints)
+unsigned emberMetadataStructureGeneration = 0;
```

Comment:
enable/disable can certainly happen enough to overflow an `unsigned`, no?  Shouldn't this be 64-bit?

Reply:
Sure, it can overflow but will we retain a cache for so long? If we enable/disable every 100ms my calculator says it takes a bit over 13.5 years to overflow and we have a single cached path/cluster pointer. 

For this to be buggy, we would need to both:
   - overflow to the exact same value (not more)
   - try to re-use the same cached path for a cluster (without any cached cluster calls in between)
   
Hence my thinking that 32-bit is sufficient. If you think lets just do 64 bit we can do that as well at a tiny ram/flash cost.

Reply:
Ah, good point, it has to overflow to the same value.  Then yeah, this is not an issue in practice.